harness/determined/cli/project.py

```diff
@@ -134,20 +134,17 @@ def describe_project(args: Namespace) -> None:
 def delete_project(args: Namespace) -> None:
     sess = setup_session(args)
     (w, p) = project_by_name(sess, args.workspace_name, args.project_name)
-    if p.numExperiments > 0:
-        raise errors.ForbiddenException(
-            authentication.must_cli_auth().get_session_user(),
-            "Projects with associated experiments currently cannot be deleted. "
-            "Use archive to hide projects.",
-        )
     if args.yes or render.yes_or_no(
         'Deleting project "' + args.project_name + '" will result in the \n'
-        "unrecoverable deletion of this project and notes. For a recoverable \n"
-        "alternative, see the 'archive' command. Do you still \n"
+        "unrecoverable deletion of this project and all of its experiments and notes.\n"
+        "For a recoverable alternative, see the 'archive' command. Do you still \n"
         "wish to proceed?"
     ):
-        bindings.delete_DeleteProject(sess, id=p.id)
-        print(f"Successfully deleted project {args.project_name}.")
+        resp = bindings.delete_DeleteProject(sess, id=p.id)
+        if resp.completed:
+            print(f"Successfully deleted project {args.project_name}.")
+        else:
+            print(f"Started deletion of project {args.project_name}...")
```

Comment:
from CLI UX perspective I'd say the CLI should be able to wait for completion.
- if waiting, report an error if it occurs.
- waiting could be optionally disabled through a flag such as `--no-wait`.